debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -94,6 +99,24 @@ protected void start(BlockingConsumer<ChangeEvent> eventConsumer, Consumer<Throw
         }
     }
 
+    private void delaySnapshotIfNeeded() {
+        Duration delay = taskContext.getConfig().getSnapshotDelay();
+        if (delay.isZero() || delay.isNegative()) {
+            return;
+        }
+
+        logger.info("The connector will wait for {} ms before proceeding", delay.toMillis());
+        Threads.Timer timer = Threads.timer(Clock.SYSTEM, delay);
+        Metronome metronome = Metronome.parker(ConfigurationDefaults.RETURN_CONTROL_INTERVAL, Clock.SYSTEM);
+        while (!timer.expired()) {
+            try {
+                metronome.pause();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
A `return` is missing here, otherwise the method will continue to run, just ignoring the request to interrupt. I'll add it.

Another issue is that `takeSnapshot()` will still be invoked, also if the executor has been requested to stop. The reason being that `CompleteableFuture` submits all the given steps early on, and `shutdown()` still lets all already submitted tasks finish. So I'll abort early in `takeSnapshot()` if we have been stopped already.